capa/features/extractors/cape/file.py

```diff
@@ -7,106 +7,98 @@
 # See the License for the specific language governing permissions and limitations under the License.
 
 import logging
-from typing import Dict, Tuple, Iterator
+from typing import Tuple, Iterator
 
 from capa.features.file import Export, Import, Section
 from capa.features.common import String, Feature
 from capa.features.address import NO_ADDRESS, Address, ProcessAddress, AbsoluteVirtualAddress
 from capa.features.extractors.helpers import generate_symbols
+from capa.features.extractors.cape.models import CapeReport
 from capa.features.extractors.base_extractor import ProcessHandle
 
 logger = logging.getLogger(__name__)
 
 
-def get_processes(static: Dict) -> Iterator[ProcessHandle]:
+def get_processes(report: CapeReport) -> Iterator[ProcessHandle]:
     """
     get all the created processes for a sample
     """
+    for process in report.behavior.processes:
+        addr = ProcessAddress(pid=process.process_id, ppid=process.parent_id)
+        yield ProcessHandle(address=addr, inner=process)
 
-    def rec(process):
-        address: ProcessAddress = ProcessAddress(pid=process["pid"], ppid=process["parent_id"])
-        inner: Dict[str, str] = {"name": process["name"]}
-        yield ProcessHandle(address=address, inner=inner)
-        for child in process["children"]:
-            yield from rec(child)
 
-    for process in static["processtree"]:
-        yield from rec(process)
-
-
-def extract_import_names(static: Dict) -> Iterator[Tuple[Feature, Address]]:
+def extract_import_names(report: CapeReport) -> Iterator[Tuple[Feature, Address]]:
     """
     extract imported function names
     """
-    imports = static["imports"]
+    assert report.static is not None and report.static.pe is not None
+    imports = report.static.pe.imports
 
-    """
-        2.2-CAPE
-        "imports": [
-            {
-             "dll": "RPCRT4.dll",
-             "imports": [{"address": "0x40504c","name": "NdrSimpleTypeUnmarshall"}, ...]
-            },
-            ...
-        ]
-
-        2.4-CAPE
-        "imports": {
-            "ADVAPI32": {
-                "dll": "ADVAPI32.dll",
-                "imports": [{"address": "0x522000", "name": "OpenSCManagerA"}, ...],
-                ...
-            },
-            ...
-        }
-    """
     if isinstance(imports, dict):
-        imports = imports.values()
+        imports = list(imports.values())
+
+    assert isinstance(imports, list)
 
     for library in imports:
-        for function in library["imports"]:
-            addr = int(function["address"], 16)
-            for name in generate_symbols(library["dll"], function["name"]):
-                yield Import(name), AbsoluteVirtualAddress(addr)
+        for function in library.imports:
+            for name in generate_symbols(library.dll, function.name):
```

Comment:
I forgot to mention this in the review, but sometimes the import's name would be null. The suggested fix I propose is to use the addresses to identify imports if there is no function name present:

```python
for library in imports:
    for function in library.imports:
        _name = function.name if function.name else str(function.address)
        for name in generate_symbols(library.dll, _name):
            yield Import(name), AbsoluteVirtualAddress(function.address)
```